main.cpp

```diff
@@ -18,13 +18,34 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+private:
+    Rewriter& rewriter_;
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : rewriter_(rewriter) {}
 
     void run(const MatchFinder::MatchResult &Result) override {
         // Your code goes here
+	const auto* CastExpr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+	  if (CastExpr ->getCastKind() == CK_ToVoid)
+            return;
+	  if (CastExpr ->getExprLoc().isMacroID())
+            return;
+	  auto ReplaceRange = CharSourceRange::getCharRange(
+      		CastExpr->getLParenLoc(), CastExpr->getSubExprAsWritten()->getBeginLoc());
+	  StringRef text = 
+		    Lexer::getSourceText(CharSourceRange::getTokenRange(
+						CastExpr->getLParenLoc().getLocWithOffset(1), 
+                				CastExpr->getRParenLoc().getLocWithOffset(-1)),
+						*Result.SourceManager, Result.Context->getLangOpts());
+	  std::string replaceText = ("static_cast<" + text + ">").str();
+          auto castIgnore = CastExpr->getSubExprAsWritten()->IgnoreImpCasts();
+	  if(!isa<ParenExpr>(castIgnore)){
+	      replaceText.push_back('(');
+	      rewriter_.InsertText(Lexer::getLocForEndOfToken(castIgnore->getEndLoc(),

```

Comment:
Please describe the reason of using the`Lexer::getLocForEndOfToken()` function here.
What would be the difference if `rewriter.InsertText(styleCastExpr->getEndLoc().getLocOffset(1), ")");` was used here?
What would be the difference if  `rewriter.InsertTextAfterToken(styleCastExpr->getEndLoc(), ")");` was used here?

Reply:
The Lexer::getLocForEndOfToken() function is a function to help jump to the end of the line, that is, before ";".

- Using rewriter.InsertText(styleCastExpr->getEndLoc().getLocOffset(1), ")"); then ")" will be inserted immediately after styleCastExpr + 1(offset), that is, if the length of the variable name is more than 1 character, this will not give us the desired result.

- Using rewriter.InsertTextAfterToken(styleCastExpr->getEndLoc(), ")"); then we will insert after the token ")", that is, we will insert after the name of the variable what we need. 

Both options Lexer::getLocForEndOfToken(styleCastExpr->getEndLoc(),...) and rewriter.InsertTextAfterToken(styleCastExpr->getEndLoc(),...) can be applied here, they give the same expected result